mathics/builtin/numbers/calculus.py

```diff
@@ -812,6 +812,193 @@ def to_sympy(self, expr, **kwargs):
         except Exception:
             return None
 
+ class Reduce(Builtin):
+    
+    """
+    <dl>
+       <dt>'Reduce[$eqs$, $vars$]'
+       <dd> Reduces the statements in $eqs$ by solving equations in $vars$ and eliminating quantifiers.
+    </dl>
+    >> Reduce[x y == 0,{x, y}]
+     = x == 0  ||   y == 0 
+    >> Reduce[{x y == 0, y == 4},{x, y}]
+     = y == 4 && x == 0
+    >> Reduce[{x y == 0, y > 4},{x, y}]
+     =  x == 0 && y > 4
+    >> Reduce[x*y>0,{x,y}]
+     = (x < 0 && y < 0) || (x > 0 && y > 0)
+    >> Reduce[{x*y > 0, y>0},{x,y}]
+     =  x > 0 && y > 0
+    
+    >> Reduce[a<b,a]
+     =  b ∈ Reals && a < b
+    """
+    
+    messages = {
+        "eqf": "`1` is not a well-formed equation.",
+        "svars": 'Equations may not give solutions for all "solve" variables.',
+    }
+
+    rules = {
+        "Reduce[eqs_, vars_, Complexes]": "Reduce[eqs, vars]",
+       
+    }
+
+    summary_text = "find generic solutions for variables, in logical expression form"
+    
+    
+    def apply(self, eqs, vars, evaluation):
+        "Reduce[eqs_, vars_]"
+        
+        vars_original = vars
+        head_name = vars.get_head_name()
+        if head_name == "System`List":
+            vars = vars.leaves
+        else:
+            vars = [vars]
+        for var in vars:
+            if (
```

Comment:
In particular, the code that follows that line could be eliminated.

Reply:
Why?

Reply:
I mean, if you handle the ``vars`` before, you do not need to do this sanity check...